actions/class.PropertiesAuthoring.php

```diff
@@ -304,12 +304,43 @@ public function getClassForm(core_kernel_classes_Class $class): tao_helpers_form
                 //save all properties values
                 if (isset($data['properties'])) {
                     $this->saveProperties($data);
+                    $this->populateSubmittedProperties($myForm, $data);
                 }
             }
         }
         return $myForm;
     }
 
+    private function populateSubmittedProperties($myForm, $data): void
+    {
+        $elementRangeArray = [];
+        $groups = $myForm->getGroups();
+        if (isset($data['properties'])) {
+            foreach ($data['properties'] as $prop) {
+                if (!empty($prop['range'])) {
+                    $elementUri = $groups['property_' . $prop['uri']]['elements'][0];
+                    if (isset($elementUri)) {
+                        $index = strstr($elementUri, '_', true);
+                        $elementRangeArray[$index . '_range_list'] = $prop['range'];
+                    }
+                }
+            }
+        }
```

Comment:
1. We can move `$grops` inside if-statement;
2. We should check that `$prop['range']` and `$prop['uri']` were provided;
3. If `$groups['property_' . $prop['uri']]['elements'][0]` not exists - we should use `?? null` for the next if-statement.

All these changes are not required, but WHAT IF...? 🙂😅 

Wdyt, @chinnu-focaloid @gabrielfs7 @boajer ?

```suggestion
        $elementRangeArray = [];

        if (isset($data['properties'])) {
            $groups = $myForm->getGroups();

            foreach ($data['properties'] as $prop) {
                if (isset($prop['range'], $prop['uri']) && !empty($prop['range'])) {
                    $elementUri = $groups['property_' . $prop['uri']]['elements'][0] ?? null;

                    if (isset($elementUri)) {
                        $index = strstr($elementUri, '_', true);
                        $elementRangeArray[$index . '_range_list'] = $prop['range'];
                    }
                }
            }
        }
```

Reply:
Done